### ISSUE TITLE
publish arguments checks are limiting the search filters alot

### ISSUE THREAD

**Issue by illmat** [aslagle/reactive-table#250]
#190 causes an exception if I'm not using a key in a fields object.
```
fields: [{
  label: '',
  tmpl: Template.toolbar
}]
```
null fails at the string check.  
It works with an empty string as key, maybe you should mention that in the docs.

**Comment by illmat** [comment]
While the first problem wasn't critical, this one is.
Some of my filter are looking like this.
```
template.filter.set({
  '$ne': aor
});```
This fails with the new "String" check.  
Furthermore, this makes it impossible to use regexp and other advanced search filter.

**Comment by aslagle** [comment]
Sorry about that, fixed in v0.8.9. I made the filters allow strings or objects, and made the key default to an empty string if it's not passed in (though I'm actually surprised it worked without a key before). Thanks for catching this.
